app/src/main/java/com/dessalines/thumbkey/ui/components/keyboard/KeyboardScreen.kt

```diff
@@ -302,7 +301,7 @@ fun KeyboardScreen(
             }
         }
     } else {
-        if (ctx.currentInputConnection.requestCursorUpdates(CURSOR_UPDATE_MONITOR or CURSOR_UPDATE_FILTER_INSERTION_MARKER)) {
```

Comment:
Is there a reason you disabled the insertion marker filter? According to https://developer.android.com/reference/android/view/inputmethod/InputConnection#CURSOR_UPDATE_IMMEDIATE, "by default all of [CURSOR_UPDATE_FILTER_EDITOR_BOUNDS](https://developer.android.com/reference/android/view/inputmethod/InputConnection#CURSOR_UPDATE_FILTER_EDITOR_BOUNDS), [CURSOR_UPDATE_FILTER_CHARACTER_BOUNDS](https://developer.android.com/reference/android/view/inputmethod/InputConnection#CURSOR_UPDATE_FILTER_CHARACTER_BOUNDS), [CURSOR_UPDATE_FILTER_VISIBLE_LINE_BOUNDS](https://developer.android.com/reference/android/view/inputmethod/InputConnection#CURSOR_UPDATE_FILTER_VISIBLE_LINE_BOUNDS), [CURSOR_UPDATE_FILTER_TEXT_APPEARANCE](https://developer.android.com/reference/android/view/inputmethod/InputConnection#CURSOR_UPDATE_FILTER_TEXT_APPEARANCE), and [CURSOR_UPDATE_FILTER_INSERTION_MARKER](https://developer.android.com/reference/android/view/inputmethod/InputConnection#CURSOR_UPDATE_FILTER_INSERTION_MARKER), are included but specifying them can filter-out others. **It can be CPU intensive to include all, filtering specific info is recommended**" (emph. added)

Reply:
Note that CURSOR_UPDATE_MONITOR isn't a filter, they do it weird where they have flags for how to monitor and filter flags in one parameter. They API34 way with separate parameters makes more sense to me, but that's how this one is

Reply:
When I had that `or CURSOR_UPDATE_FILTER_INSERTION_MARKER`, it failed everywhere.

Reply:
that's weird, even in signal?

Reply:
I'd need to test again, but I'm pretty sure.